diagnosis/getkuberneteslogs.sh

```diff
@@ -23,6 +23,151 @@ function restore_ssh_config
     fi
 }
 
+retrycmd_if_failure() 
+{ 
+    retries=$1; 
+    wait=$2; 
+    for i in $(seq 1 $retries); do 
+        ${@:3}; [ $? -eq 0  ] && break || sleep $wait; 
+    done; 
+    log_level -i "Command Executed $i times."; 
+}
+
+
+function login_azs()
+{
+    local spn_id=$1
+    local spn_secret=$2
+    local tenant_id=$3
+    
+    az login --service-principal -u $spn_id -p $spn_secret --tenant $tenant_id
+}
+
+askSubscription() {
+    az account list -otable
+    echo ""
+    echo "Please enter the Id of the account you wish to use. If you do not see"
+    echo "a valid account in the list press Ctrl+C to abort and create one."
+    echo "If you leave this blank we will use the Current account."
+    echo -n "> "
+    read azure_subscription_id
+
+    if [ "$azure_subscription_id" != "" ]; then
+        az account set --subscription $azure_subscription_id
+    else
+        azure_subscription_id=$(az account list --output json | jq -r '.[] | select(.isDefault==true) | .id')
+    fi
+    echo "Using subscription_id: $azure_subscription_id"
+}
+
+askName() {
+    echo ""
+    echo "Choose a name for your resource group, storage account and client"
+    echo "client. This is arbitrary, but it must not already be in use by"
+    echo "any of those resources. ALPHANUMERIC ONLY. Ex: mypackerbuild"
+    echo -n "> "
+    read meta_name
+}
+
+createResourceGroup() {
+    local location=$1
+    echo "==> Creating resource group"
```

Comment:
Update prefix please (`==>`), the script is already using prefix `echo "[$(date +%Y%m%d%H%M%S)][XXX]`